File: 1491-average-salary-excluding-the-minimum-and-maximum-salary/1491-average-salary-excluding-the-minimum-and-maximum-salary.cpp
class Solution {
public:
    double average(vector<int>& salary) {
        double min=salary[0], max=0, sum=0;
        int n = salary.size();
        for(int i=0 ; i<n ; i++){
            min = min<salary[i]?min : salary[i];
            max = max>salary[i]?max : salary[i];
            sum+=salary[i];
        }
        return (sum-min-max)/(n-2);
    }
};

File: 1491-average-salary-excluding-the-minimum-and-maximum-salary/1491-average-salary-excluding-the-minimum-and-maximum-salary.java
class Solution {
    public double average(int[] salary) {
        double res = 0;
        int mi = salary[0], ma = salary[1];
        for(int ele: salary){
            mi = mi > ele? ele: mi;
            ma = ma > ele? ma: ele;
            res += ele;
        }
        res = res - ma - mi;
        res = res/(salary.length-2);
        return res;
    }
}